NICE.Identity.Authentication.Sdk/TokenStore/RedisApiTokenStore.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Threading.Tasks;
+using NICE.Identity.Authentication.Sdk.Domain;
+using Newtonsoft.Json;
+using StackExchange.Redis.Extensions.Core;
+using StackExchange.Redis.Extensions.Newtonsoft;
+#if !NET452
+using StackExchange.Redis.Extensions.Core.Implementations;
+using StackExchange.Redis.Extensions.Core.Abstractions;
+using StackExchange.Redis.Extensions.Core.Configuration;
+#endif
+
+namespace NICE.Identity.Authentication.Sdk.TokenStore
+{
+    public class RedisApiTokenStore : IApiTokenStore, IDisposable
+    {
+        private const string KeyPrefix = "APITOKEN-";
+#if NETSTANDARD2_0 || NETCOREAPP3_1 || NET461
+        private readonly IRedisDatabase _cacheClient;
+        private readonly IRedisCacheConnectionPoolManager _connectionPoolManager;
+#else
+        private readonly StackExchangeRedisCacheClient _cacheClient;
+#endif
+
+#if NETSTANDARD2_0 || NETCOREAPP3_1
+        public RedisApiTokenStore(IRedisCacheClient cacheClient)
+        {
+            _cacheClient = cacheClient.GetDbFromConfiguration();
+        }
+#elif NET461
+        public RedisApiTokenStore(string redisConnectionString)
+        {
+            var serializer = new NewtonsoftSerializer(new JsonSerializerSettings
+            {
+                ReferenceLoopHandling = ReferenceLoopHandling.Ignore
```

Comment:
why do you need this?

Reply:
This is to ignore errors from reference loops when doing serialisation. It shouldn't be an issue due to the token class structure. It came from the code in RedisOwinSessionStore, but I suppose we can remove it from here.